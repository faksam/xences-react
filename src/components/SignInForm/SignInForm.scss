.sign-in-form {
  .input-field {
    &.col {
      padding: 0;
      float: none;
    }
  }
}
