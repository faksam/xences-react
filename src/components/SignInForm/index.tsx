// import libraries
import * as React from 'react';

// third-party libraries

// styles
import './signInForm.scss';

// components
import { Button, Col, Row, TextInput } from 'react-materialize';

const SignInForm = () => (
  <Row className="signInForm">
    <Col s={6} >
      {/* <TextInput password label="Password" /> */}
      <TextInput className="formInput" email validate label="Email" />
      <TextInput className="formInput" password label="Password" />
      <Button className="">Sign In</Button>
    </Col>
  </Row>
);

export default SignInForm;
