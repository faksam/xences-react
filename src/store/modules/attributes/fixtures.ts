export const attributes = [
  {
    id: '1',
    name: 'color',
    description: 'Product color',
    createdAt:  '12/05/19',
  },
  {
    id: '2',
    name: 'size',
    description: 'Product size',
    createdAt:  '12/05/19',
  },
  {
    id: '3',
    name: 'color',
    description: 'Product color',
    createdAt:  '12/05/19',
  },
];
