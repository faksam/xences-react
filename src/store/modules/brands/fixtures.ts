export const brands = [
  {
    id: '1',
    name: 'Nike',
    description: 'Brand Nike',
    createdAt:  '12/05/19',
  },
  {
    id: '2',
    name: 'Puma',
    description: 'Brand Puma',
    createdAt:  '12/05/19',
  },
  {
    id: '3',
    name: 'Otunba',
    description: 'Brand Otunba',
    createdAt:  '12/05/19',
  },
];

export const colors = [
  {
    id: '1',
    name: 'Black',
    description: 'Black',
    createdAt:  '12/05/19',
  },
  {
    id: '2',
    name: 'Red',
    description: 'Red',
    createdAt:  '12/05/19',
  },
  {
    id: '3',
    name: 'Pink',
    description: 'Pink',
    createdAt:  '12/05/19',
  },
  {
    id: '4',
    name: 'White',
    description: 'White',
    createdAt:  '12/05/19',
  },
  {
    id: '5',
    name: 'Blue',
    description: 'Blue',
    createdAt:  '12/05/19',
  },
];
