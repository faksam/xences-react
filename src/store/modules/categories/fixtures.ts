export const categories = [
  {
    id: '1',
    name: 'Cat 1',
    description: 'Category One',
    createdAt:  '12/05/19',
  },
  {
    id: '2',
    name: 'Cat 2',
    description: 'Category Two',
    createdAt:  '12/05/19',
  },
  {
    id: '3',
    name: 'Cat 3',
    description: 'Category Three',
    createdAt:  '12/05/19',
  },
];
