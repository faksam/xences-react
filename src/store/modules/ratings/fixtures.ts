export const ratings = [
  {
    id: '1',
    rating: '4',
    createdAt:  '12/05/19',
  },
  {
    id: '2',
    rating: '4',
    createdAt:  '12/05/19',
  },
  {
    id: '3',
    rating: '4',
    createdAt:  '12/05/19',
  },
];
