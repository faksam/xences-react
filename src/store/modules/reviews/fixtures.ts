export const reviews = [
  {
    id: '1',
    review: 'Good product',
    createdAt:  '12/05/19',
  },
  {
    id: '2',
    review: 'Good product',
    createdAt:  '12/05/19',
  },
  {
    id: '3',
    review: 'Good product',
    createdAt:  '12/05/19',
  },
];
