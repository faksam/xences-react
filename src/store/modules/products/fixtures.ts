// fixtures
import { attributes } from '../../modules/attributes/fixtures';
import { brands } from '../../modules/brands/fixtures';
import { categories } from '../../modules/categories/fixtures';
import { ratings } from '../../modules/ratings/fixtures';
import { reviews } from '../../modules/reviews/fixtures';
import { shops } from '../../modules/shops/fixtures';

export const products = [
  {
    attributes,
    reviews,
    ratings,
    id: '1',
    name: 'Nice shirt',
    discountPercent: '10',
    price: '2000',
    gender: 'male',
    brand: brands[0],
    quantity: 7,
    pictures: 'https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_005_shaco9.jpg',
    shop: shops[0],
    category: categories[1].name,
    createdAt: '05/12/19',
  },
  {
    attributes,
    reviews,
    ratings,
    id: '2',
    name: 'Nice cap',
    discountPercent: '10',
    price: '2000',
    gender: 'male',
    brand: brands[0],
    quantity: 7,
    pictures: 'https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_005_shaco9.jpg',
    shop: shops[0],
    category: categories[1].name,
    createdAt: '05/12/19',
  },
  {
    attributes,
    reviews,
    ratings,
    id: '3',
    name: 'Nice coat',
    discountPercent: '10',
    price: '2000',
    gender: 'male',
    brand: brands[0],
    quantity: 7,
    pictures: 'https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_005_shaco9.jpg',
    shop: shops[0],
    category: categories[1].name,
    createdAt: '05/12/19',
  },
  {
    attributes,
    reviews,
    ratings,
    id: '4',
    name: 'Nice top',
    discountPercent: '10',
    price: '2000',
    gender: 'male',
    brand: brands[0],
    quantity: 7,
    pictures: 'https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_005_shaco9.jpg',
    shop: shops[0],
    category: categories[1].name,
    createdAt: '05/12/19',
  },
  {
    attributes,
    reviews,
    ratings,
    id: '5',
    name: 'Nice skirt',
    discountPercent: '10',
    price: '2000',
    gender: 'male',
    brand: brands[0],
    quantity: 7,
    pictures: 'https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_005_shaco9.jpg',
    shop: shops[0],
    category: categories[1].name,
    createdAt: '05/12/19',
  },
  {
    attributes,
    reviews,
    ratings,
    id: '6',
    name: 'Nice shirt',
    discountPercent: '10',
    price: '2000',
    gender: 'male',
    brand: brands[0],
    quantity: 7,
    pictures: 'https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_001_xaepru.jpg',
    shop: shops[0],
    category: categories[1].name,
    createdAt: '05/12/19',
  },
  {
    attributes,
    reviews,
    ratings,
    id: '7',
    name: 'Nice cap',
    discountPercent: '10',
    price: '2000',
    gender: 'male',
    brand: brands[0],
    quantity: 7,
    pictures: 'https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_001_xaepru.jpg',
    shop: shops[0],
    category: categories[1].name,
    createdAt: '05/12/19',
  },
  {
    attributes,
    reviews,
    ratings,
    id: '8',
    name: 'Nice coat',
    discountPercent: '10',
    price: '2000',
    gender: 'male',
    brand: brands[0],
    quantity: 7,
    pictures: 'https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_001_xaepru.jpg',
    shop: shops[0],
    category: categories[1].name,
    createdAt: '05/12/19',
  },
  {
    attributes,
    reviews,
    ratings,
    id: '9',
    name: 'Nice top',
    discountPercent: '10',
    price: '2000',
    gender: 'male',
    brand: brands[0],
    quantity: 7,
    pictures: 'https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_001_xaepru.jpg',
    shop: shops[0],
    category: categories[1].name,
    createdAt: '05/12/19',
  },
  {
    attributes,
    reviews,
    ratings,
    id: '10',
    name: 'Nice skirt',
    discountPercent: '10',
    price: '2000',
    gender: 'male',
    brand: brands[0],
    quantity: 7,
    pictures: 'https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_001_xaepru.jpg',
    shop: shops[0],
    category: categories[1].name,
    createdAt: '05/12/19',
  },
];

export const menClothings = [
  {
    id: '1',
    name: 'Shirts',
    description: 'Shirts',
    createdAt:  '12/05/19',
  },
  {
    id: '2',
    name: 'T-Shirts',
    description: 'T-Shirts',
    createdAt:  '12/05/19',
  },
  {
    id: '3',
    name: 'Polo Shirts',
    description: 'Polo Shirts',
    createdAt:  '12/05/19',
  },
  {
    id: '4',
    name: 'Pants',
    description: 'Pants',
    createdAt:  '12/05/19',
  },
  {
    id: '5',
    name: 'Jeans',
    description: 'Jeans',
    createdAt:  '12/05/19',
  },
  {
    id: '6',
    name: 'Jumpers - Cardigans',
    description: 'Jumpers - Cardigans',
    createdAt:  '12/05/19',
  },
  {
    id: '7',
    name: 'Hoodies - Sweatshirts',
    description: 'Hoodies - Sweatshirts',
    createdAt:  '12/05/19',
  },
  {
    id: '8',
    name: 'Blazers - Suits',
    description: 'Blazers - Suits',
    createdAt:  '12/05/19',
  },
  {
    id: '9',
    name: 'Jackets - Coats',
    description: 'Jackets - Coats',
    createdAt:  '12/05/19',
  },
  {
    id: '10',
    name: 'Underwear - Sleepwear',
    description: 'Underwear - Sleepwear',
    createdAt:  '12/05/19',
  },
  {
    id: '10',
    name: 'Beachwear',
    description: 'Beachwear',
    createdAt:  '12/05/19',
  },
];
