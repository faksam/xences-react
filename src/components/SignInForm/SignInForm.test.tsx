// react libraries
import * as React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';

// third-party libraries
import { mount } from 'enzyme';

// components
import SignInForm from './';

// fixtures
import { products } from '../../store/modules/products/fixtures';

describe('The SignInForm component', () => {
  describe('Successful displays', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = mount(
        <Router>
          <SignInForm />
        </Router>
      );
    });

    it('should be render products page', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
