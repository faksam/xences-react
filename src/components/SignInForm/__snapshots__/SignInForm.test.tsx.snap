// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The SignInForm component Successful displays should be render products page 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <SignInForm>
      <Row
        className="sign-in-form"
      >
        <div
          className="row sign-in-form"
        >
          <Col
            s={6}
          >
            <div
              className="col s6"
            >
              <TextInput
                className="formInput"
                email={true}
                label="Email"
                validate={true}
              >
                <div
                  className="input-field col"
                >
                  <input
                    className="formInput"
                    id="email3"
                    type="email"
                  />
                  <label
                    className=""
                    htmlFor="email3"
                  >
                    Email
                  </label>
                  <span
                    className="helper-text"
                  />
                </div>
              </TextInput>
              <TextInput
                className="formInput"
                label="Password"
                password={true}
              >
                <div
                  className="input-field col"
                >
                  <input
                    className="formInput"
                    id="password5"
                    type="password"
                  />
                  <label
                    className=""
                    htmlFor="password5"
                  >
                    Password
                  </label>
                  <span
                    className="helper-text"
                  />
                </div>
              </TextInput>
              <Button
                className=""
                node="button"
              >
                <button
                  className="btn"
                  disabled={false}
                >
                  Sign In
                </button>
              </Button>
            </div>
          </Col>
        </div>
      </Row>
    </SignInForm>
  </Router>
</BrowserRouter>
`;
